pkg/services/ngalert/schedule/schedule.go

```diff
@@ -139,18 +154,164 @@ func (sch *schedule) Unpause() error {
 }
 
 func (sch *schedule) Run(ctx context.Context) error {
-	sch.wg.Add(1)
+	sch.wg.Add(2)
 
 	go func() {
 		if err := sch.ruleEvaluationLoop(ctx); err != nil {
 			sch.log.Error("failure while running the rule evaluation loop", "err", err)
 		}
 	}()
 
+	go func() {
```

Comment:
Ideally we should have `defer sch.wg.Done()` inside this goroutine here and not inside the below function call to make it more readable and easy to debug. Same for `ruleEvaluationLoop` above. Hence have the `wg` as a local variable here and not in `*schedule`. WDYT?

Reply:
Done!